bb_wrapper/wrapper/bb.py

```diff
@@ -101,8 +115,37 @@ def authenticate(self):
         }
         kwargs = dict(headers=header, verify=False, data=data)
 
-        response = requests.post(url, **kwargs)
-        response = self._process_response(response)
-        self.__access_token = response.data["access_token"]
-        self.__token_type = response.data["token_type"]
+        if self.data.access_token is None or force_auth is True:
```

Comment:
Seria algo nesse estilo
```suggestion
        elapsed_time = time.time() - self.data.token_time
        if elapsed_time >= self.TOKEN_TIME:
            force_auth = True
        if self.data.access_token is None or force_auth is True:
```